aws/rust-runtime/aws-config/src/sts/assume_role.rs

```diff
@@ -459,6 +458,56 @@ mod test {
         assert_eq!("https://sts-fips.us-west-17.api.aws/", req.uri())
     }
 
+    #[tokio::test]
+    async fn configures_fips_endpoint_from_env() {
+        let (server, request) = capture_request(None);
+        let sdk_config = crate::from_env()
+            .region(Region::from_static("us-east-1"))
+            .env(Env::from_slice(&[("AWS_USE_FIPS_ENDPOINT", "true")]))
+            .http_connector(server)
+            .load()
+            .await;
+        assert!(sdk_config.use_fips().unwrap_or_default());
+        let provider = AssumeRoleProvider::builder("myrole")
+            .configure(&sdk_config)
+            .build_from_provider(provide_credentials_fn(|| async {
+                Ok(Credentials::for_tests())
+            }))
+            .await;
+        let _ = provider.provide_credentials().await;
+        let req = request.expect_request();
+        assert!(req.uri().host().unwrap().contains("sts-fips"));
+    }
```

Comment:
these tests are basically equivalent to the tests that already existed—we need tests of the profile file provider / default chain itself

Reply:
fixed.